From 3a812de6e7e0b285eda21ebbb670ca809c4db2b5 Mon Sep 17 00:00:00 2001
From: Matt Jolly <Matt.Jolly@footclan.ninja>
Date: Thu, 11 Apr 2024 12:42:17 +1000
Subject: [PATCH] Add QtGui to the list of Qt modules

Re: https://bugreports.qt.io/browse/QTBUG-124135 - certain versions of QT
(6.6.3, 6.7.0) produce pkg-config files that no longer emit dependencies.
This causes Chromium builds to fail as due to not seeing QtGui as a dependency
of QtWidgets:

    `gen/qt6/../../../../ui/qt/qt_shim.h:11:10: fatal error: 'QImage' file not found`

While this has been resolved upstream, there will certainly be some users
that are still using these versions of QT. This patch adds QtGui to the list of
Qt modules that are required to build the Chromium UI.

Bug: https://bugs.gentoo.org/928299
Suggested-by: Chris Pritchard <chris@christopherpritchard.co.uk>
Signed-off-by: Matt Jolly <kangie@gentoo.org>
--- a/ui/qt/BUILD.gn
+++ b/ui/qt/BUILD.gn
@@ -60,6 +60,7 @@ template("qt_shim") {
   pkg_config("qt" + invoker.qt_version + "_config") {
     packages = [
       "Qt" + invoker.qt_version + "Core",
+      "Qt" + invoker.qt_version + "Gui",
       "Qt" + invoker.qt_version + "Widgets",
     ]
   }
-- 
2.44.0
